src/main/steps/applicant/statement-of-service/choose-parties/postController.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { handleError, removeUploadDocErrors } from '../../../../../main/steps/common/documents/upload/utils';
+import { applyParms } from '../../../../../main/steps/common/url-parser';
+import { APPLICANT_STATEMENT_OF_SERVICE, APPLICANT_STATEMENT_OF_SERVICE_SUMMARY } from '../../../../../main/steps/urls';
+import { CosApiClient } from '../../../../app/case/CosApiClient';
+import { CaseWithId } from '../../../../app/case/case';
+import { AppRequest } from '../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../app/controller/PostController';
+import { Form, FormError, FormFields, FormFieldsFn } from '../../../../app/form/Form';
+
+@autobind
+export default class UploadSosPostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  private setRedirectUrl(req: AppRequest<Partial<CaseWithId>>) {
+    return applyParms(APPLICANT_STATEMENT_OF_SERVICE, {
+      context: req.params.context,
+    });
+  }
+
+  private async uploadDocument(req: AppRequest, res: Response): Promise<void> {
+    const { session, files } = req;
+    const { user } = session;
+    const client = new CosApiClient(user.accessToken, req.locals.logger);
+    const redirectUrl = this.setRedirectUrl(req);
+
+    req.url = redirectUrl;
+
+    if (!files) {
+      req.session.errors = handleError(req.session.errors, 'empty');
+      return this.redirect(req, res, redirectUrl);
+    }
+
+    try {
+      if (req.session.userCase['statementOfServiceDocument']) {
```

Comment:
access properties with . not using [''].

Reply:
Changed usage